homeassistant/components/lock/kiwi.py

```diff
@@ -0,0 +1,166 @@
+"""
+Support for the KIWI.KI lock platform.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/lock.kiwi/
+"""
+import logging
+import datetime
+import requests
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.lock import (LockDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_USERNAME, ATTR_ID, ATTR_LONGITUDE, ATTR_LATITUDE)
+
+REQUIREMENTS = ['python-dateutil==2.6.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_TYPE = "hardware type"
+ATTR_PERMISSION = 'permission'
+ATTR_CAN_INVITE = "can invite others"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+BASE_URL = 'https://api.kiwi.ki'
+API_AUTH_URL = BASE_URL + '/pre/session/'
+API_LIST_DOOR_URL = BASE_URL + '/pre/sensors/'
+API_OPEN_DOOR_URL = BASE_URL + '/pre/sensors/{}/act/open'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the KIWI lock platform."""
+    kiwi = KiwiClient(config.get(CONF_USERNAME), config.get(CONF_PASSWORD))
+    add_devices([KiwiLock(lock, kiwi) for lock in kiwi.get_locks()], True)
+
+
+class KiwiClient:
```

Comment:
This code should be placed in a module which can be installed from PyPI. This way other people can re-use it and we don't need to ship device-specific code with Home Assistant.

Reply:
That would be ideal but If you look at it - it doesn't do much but and in my opinion it's not worth creating a PyPI package for it. I reference to the [_locktrion_ lock](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/lock/lockitron.py) which does not use a site package as well. I'm pro moving to a PyPI once functionality grows.

Reply:
https://home-assistant.io/developers/code_review_platform/

See 6.1 and the note on top.

Reply:
OK I'll create a pip. Just give me some days to finish my vacation 🏄